node/node.go

```diff
@@ -1,1356 +1,15 @@
-	_, err = pvsc.GetPubKey(context.TODO())
-	if err != nil {
-		return nil, fmt.Errorf("can't get pubkey: %w", err)
-	}
-
-	const (
-		retries = 50 // 50 * 100ms = 5s total
-		timeout = 100 * time.Millisecond
-	)
-	pvscWithRetries := privval.NewRetrySignerClient(pvsc, retries, timeout)
-
-	return pvscWithRetries, nil
-}
-
-func createAndStartPrivValidatorGRPCClient(
-	config *cfg.Config,
-	chainID string,
-	logger log.Logger,
-) (types.PrivValidator, error) {
-	pvsc, err := tmgrpc.DialRemoteSigner(config, chainID, logger)
-	if err != nil {
-		return nil, fmt.Errorf("failed to start private validator: %w", err)
-	}
-
-	// try to get a pubkey from private validate first time
-	_, err = pvsc.GetPubKey(context.TODO())
-	if err != nil {
-		return nil, fmt.Errorf("can't get pubkey: %w", err)
-	}
-
-	return pvsc, nil
-}
-
-func getRouterConfig(conf *cfg.Config, proxyApp proxy.AppConns) p2p.RouterOptions {
-	opts := p2p.RouterOptions{
-		QueueType: conf.P2P.QueueType,
-	}
-
-	if conf.P2P.MaxNumInboundPeers > 0 {
-		opts.MaxIncomingConnectionAttempts = conf.P2P.MaxIncomingConnectionAttempts
-	}
-
-	if conf.FilterPeers && proxyApp != nil {
-		opts.FilterPeerByID = func(ctx context.Context, id p2p.NodeID) error {
-			res, err := proxyApp.Query().QuerySync(context.Background(), abci.RequestQuery{
-				Path: fmt.Sprintf("/p2p/filter/id/%s", id),
-			})
-			if err != nil {
-				return err
-			}
-			if res.IsErr() {
-				return fmt.Errorf("error querying abci app: %v", res)
-			}
-
-			return nil
-		}
-
-		opts.FilterPeerByIP = func(ctx context.Context, ip net.IP, port uint16) error {
-			res, err := proxyApp.Query().QuerySync(ctx, abci.RequestQuery{
-				Path: fmt.Sprintf("/p2p/filter/addr/%s", net.JoinHostPort(ip.String(), strconv.Itoa(int(port)))),
-			})
-			if err != nil {
-				return err
-			}
-			if res.IsErr() {
-				return fmt.Errorf("error querying abci app: %v", res)
-			}
-
-			return nil
-		}
-
-	}
-
-	return opts
-}
-
-// FIXME: Temporary helper function, shims should be removed.
-func makeChannelsFromShims(
-	router *p2p.Router,
-	chShims map[p2p.ChannelID]*p2p.ChannelDescriptorShim,
-) map[p2p.ChannelID]*p2p.Channel {
-
-	channels := map[p2p.ChannelID]*p2p.Channel{}
-	for chID, chShim := range chShims {
-		ch, err := router.OpenChannel(*chShim.Descriptor, chShim.MsgType, chShim.Descriptor.RecvBufferCapacity)
-		if err != nil {
-			panic(fmt.Sprintf("failed to open channel %v: %v", chID, err))
-		}
-
-		channels[chID] = ch
-	}
-
-	return channels
-}
-
-func getChannelsFromShim(reactorShim *p2p.ReactorShim) map[p2p.ChannelID]*p2p.Channel {
-	channels := map[p2p.ChannelID]*p2p.Channel{}
-	for chID := range reactorShim.Channels {
-		channels[chID] = reactorShim.GetChannel(chID)
-	}
-
-	return channels
+// New constructs a tendermint node service for use in go process that
+// host their own process-local tendermint node.
+func New(conf *config.Config, logger log.Logger) (service.Service, error) {
```

Comment:
Having these arguments to set up a node means that users that run Tendermint programatically can't use custom-reactors (or a different state-provider but that's probably less of a concern). I'm not saying that we shouldn't want this, I just think we need to be deliberate in the decision when it comes to allowing / disallowing custom reactors

Reply:
Perhaps more important than this is that it looks like with this configuration that nodes won't be able to run Tendermint in-process. We should have something like  a `proxy.ClientCreator` or a way of optionally passing the `abci.Application` interface when applications aren't using a socket.